site/app/controllers/forum/ForumController.php

```diff
@@ -775,15 +795,20 @@ public function getHistory(){
         return $this->core->getOutput()->getOutput();
     }
 
+    public function modifyAnonymous($author) {
+        return $this->core->getUser()->accessFullGrading() || $this->core->getUser()->getId() === $author;
+    }
+
     public function getEditPostContent(){
         $post_id = $_POST["post_id"];
         if($this->checkPostEditAccess($post_id) && !empty($post_id)) {
             $result = $this->core->getQueries()->getPost($post_id);
             $output = array();
-            $output['user'] = $result["author_user_id"];
             $output['post'] = $result["content"];
             $output['post_time'] = $result['timestamp'];
             $output['anon'] = $result['anonymous'];
+            $output['change_anon'] = $this->modifyAnonymous($result["author_user_id"]);
+            $output['user'] = $output['anon'] ? 'Anonymous' : $result["author_user_id"];
```

Comment:
If a post is marked as anonymous, should the edit banner always display anonymous? For limited access graders this should always be the case. But for the author, ta's and instructors should this display anonymous?